### PR TITLE
Issue specific line types for added/deleted end of file newlines.

### LINKED ISSUE #11853: Issue separate types for ending newline related changes
Required for https://github.com/mozilla/addons-code-manager/issues/877

We am rewriting all end of file newline change flags (GIT_DIFF_LINE_CONTEXT_EOFNL, GIT_DIFF_LINE_ADD_EOFNL, GIT_DIFF_LINE_DEL_EOFNL) to be `normal`, `insert`, `delete` instead. So there's context being lost which creates a few issues.

Let's implement separate `normal-eofnl`, `insert-eofnl`, `delete-eofnl` types for these lines instead.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #11853

### CODE REVIEW

File: src/olympia/lib/tests/test_git.py [hunk 1]
```diff
@@ -873,18 +903,65 @@ def test_get_diff_newline_both_no_newline():
...
+            'old_line_number': -1,
+            'type': 'insert'
+        },
+        {
+            'content': '\n\\ No newline at end of file',
+            'new_line_number': 2,
+            'old_line_number': -1,
+            'type': 'delete-eofnl'
+        }
+    ]
```
Comment (EnTeQuAk) [inline src/olympia/lib/tests/test_git.py]: I added a comment and tested this twice… that's indeed the hunks that represent the above diff, I don't quite know or understand why the newline changes aren't just `GIT_DIFF_LINE_CONTEXT_EOFNL` / `normal-eofnl` as they didn't change, but I haven't managed to create a scenario where that would be the case.

The `new_line_number` and `old_line_number` also change accordingly to the line they were referring to.

@bobsilverberg 	that's most certainly something to be aware of.

**Comment by EnTeQuAk** [comment]
As a side note, I'll file an issue that tries to document the whole diff structure in the api docs as well - to document these kinds of specifics (what types are possible values etc). It's beyond this PR though I think.

File: src/olympia/lib/git.py [hunk 2]
```diff
@@ -52,6 +52,9 @@
     GIT_DIFF_LINE_CONTEXT: 'normal',
     GIT_DIFF_LINE_ADDITION: 'insert',
     GIT_DIFF_LINE_DELETION: 'delete',
+    GIT_DIFF_LINE_CONTEXT_EOFNL: 'normal-eofnl',
+    GIT_DIFF_LINE_ADD_EOFNL: 'insert-eofnl',
+    GIT_DIFF_LINE_DEL_EOFNL: 'delete-eofnl',
```
Comment (eviljeff) [inline src/olympia/lib/git.py]: Where are these new properties referenced?

Comment (EnTeQuAk) [reply src/olympia/lib/git.py]: https://github.com/mozilla/addons-server/pull/11912/files#diff-c58b3b043d425ab67d30f4ad957a3b3eR44-R48 defines them and https://github.com/mozilla/addons-server/pull/11912/files#diff-c58b3b043d425ab67d30f4ad957a3b3eR597 actually makes use of them by passing on the `origin` and looking it up in this mapping.

**Comment by bobsilverberg** [comment]
@EnTeQuAk I'm probably not going to get a chance to look at this until Friday. If that's a problem I think it's probably fine to merge this and I can open a new issue if I find any problems with the API response.

**Comment by EnTeQuAk** [comment]
> @EnTeQuAk I'm probably not going to get a chance to look at this until Friday. If that's a problem I think it's probably fine to merge this and I can open a new issue if I find any problems with the API response.

I'm happy to wait till Friday, just let me know when this can be merged.

**Comment by EnTeQuAk** [comment]
> 16:40 <@kumar> cgrebs: I can help test the integration of https://github.com/mozilla/addons-server/pull/11912 into the frontend while bsilverberg is out. Since this isn't a high priority issue, no rush. I'd say just land your patch so we can test it on the frontend more easily :D

**Pull request merged**